### DISCUSSION TITLE
Ha Core su Docker Problemi caddy

### DISCUSSION THREAD

**Discussion by giacgiac** [indomus/forum#2083] [HUB: Home Assistant]
Buongiorno a tutti, ho seguito passo passo la guida, aperto le porte 443 e 9000, utilizzato il server di prova per i certificati che mi danno Obtained la prima volta, poi, una volta cambiato e tolte le prime 4 righe, facendo il log mi riporta questo:

> {"level":"info","ts":1668598554.2458913,"msg":"shutting down apps, then terminating","signal":"SIGTERM"}
> {"level":"warn","ts":1668598554.2459843,"msg":"exiting; byeee!! 👋","signal":"SIGTERM"}
> {"level":"info","ts":1668598554.2464614,"logger":"tls.cache.maintenance","msg":"stopped background certificate maintenance","cache":"0x40004aa8c0"}
> {"level":"info","ts":1668598554.2466786,"logger":"admin","msg":"stopped previous server","address":"localhost:2019"}
> {"level":"info","ts":1668598554.2467096,"msg":"shutdown complete","signal":"SIGTERM","exit_code":0}
> {"level":"info","ts":1668598555.3130226,"msg":"using provided configuration","config_file":"/etc/caddy/Caddyfile","config_adapter":"caddyfile"}
> {"level":"warn","ts":1668598555.316553,"msg":"Caddyfile input is not formatted; run the 'caddy fmt' command to fix inconsistencies","adapter":"caddyfile","file":"/etc/caddy/Caddyfile","line":2}
> {"level":"info","ts":1668598555.3188334,"logger":"admin","msg":"admin endpoint started","address":"localhost:2019","enforce_origin":false,"origins":["//localhost:2019","//[::1]:2019","//127.0.0.1:2019"]}
> {"level":"info","ts":1668598555.3196456,"logger":"http","msg":"enabling automatic HTTP->HTTPS redirects","server_name":"srv0"}
> {"level":"info","ts":1668598555.319693,"logger":"tls.cache.maintenance","msg":"started background certificate maintenance","cache":"0x40004d8f50"}
> {"level":"info","ts":1668598555.3207102,"logger":"tls","msg":"cleaning storage unit","description":"FileStorage:/data/caddy"}
> {"level":"info","ts":1668598555.322774,"logger":"tls","msg":"finished cleaning storage units"}
> {"level":"info","ts":1668598555.323243,"logger":"http","msg":"enabling HTTP/3 listener","addr":":9000"}
> {"level":"info","ts":1668598555.3235137,"msg":"failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 2048 kiB, got: 416 kiB). See https://github.com/lucas-clemente/quic-go/wiki/UDP-Receive-Buffer-Size for details."}
> {"level":"info","ts":1668598555.3239484,"logger":"http.log","msg":"server running","name":"srv0","protocols":["h1","h2","h3"]}
> {"level":"info","ts":1668598555.3244162,"logger":"http.log","msg":"server running","name":"remaining_auto_https_redirects","protocols":["h1","h2","h3"]}
> {"level":"info","ts":1668598555.3244455,"logger":"http","msg":"enabling automatic TLS certificate management","domains":["***********.duckdns.org"]}
> {"level":"info","ts":1668598555.327076,"msg":"autosaved config (load with --resume flag)","file":"/config/caddy/autosave.json"}
> {"level":"info","ts":1668598555.3271496,"msg":"serving initial configuration"}

nella directory ho comunque i file:

> total 24
> -rw------- 1 root root  227 Nov 16 12:21 ********.duckdns.org.key
> -rw------- 1 root root  161 Nov 16 12:21 ********.duckdns.org.json
> -rw------- 1 root root 5337 Nov 16 12:21 ********.duckdns.org.crt
> drwx------ 3 root root 4096 Nov 16 12:21 ..
> drwx------ 2 root root 4096 Nov 16 12:21 .

in configuration.yaml:

> http:
>   use_x_forwarded_for: true
>   trusted_proxies: 
>     - 172.17.0.0/16
>     - 127.0.0.0/8
>     - '::1'

Ma da esterno in https nada... solo http..., qualcuno può aiutarmi?

**Comment by snakuzzo** [comment]
fai vedere il file `Caddyfile`

**Reply by giacgiac** [reply]
Eccolo, grazie:

```
(https_header) {
  header {
    Strict-Transport-Security "max-age=31536000; includeSubdomains"
    X-XSS-Protection "1; mode=block"
    X-Content-Type-Options "nosniff"
    X-Frame-Options "SAMEORIGIN"
    Referrer-Policy "same-origin"
  }
}
 
https://********.duckdns.org:9000 {
  import https_header
  reverse_proxy http://192.168.1.61:8123
}
```

**Reply by snakuzzo** [reply]
formatta bene con il markdown, grazie

**Reply by giacgiac** [reply]
a saper come...

**Reply by snakuzzo** [reply]
Leggi [QUI](https://github.com/indomus/forum/discussions/1311)

**Reply by giacgiac** [reply]
forse ho capito

**Reply by giacgiac** [reply]
avevo già cercato, grazie!

**Reply by snakuzzo** [reply]
Ad ogni modo quell'errore che riscontri nel log non l'ho mai visto.
Cercando ho trovato [QUESTO](https://forum.storj.io/t/linux-failed-to-sufficiently-increase-receive-buffer-size/12494/12)
Curiosità...che problema hai quando provi ad accedere dall'esterno ? Dire "NON FUNZIONA" non basta.

**Reply by giacgiac** [reply]
Se punto su https://*******.duckdns.org:9000 mi risponde "Safari non può aprire la pagina perché la connessione sicura al server non è riuscita", provando da telefono in rete mobile, chiaramente da rete interna uguale

**Reply by giacgiac** [reply]
in http va bene

**Reply by snakuzzo** [reply]
Allora hai sbagliato tutto...hai fatto casino da qualche parte.
E' impossibile che dall'esterno funzioni in HTTP se su Caddy hai impostato che risponde in HTTPS.

**Reply by giacgiac** [reply]
ho rifatto tutto da capo decine di volte...

**Reply by snakuzzo** [reply]
...probabilmente sbagliando sempre però (almeno leggendo quello che ti capita)
Ricapitolando...

devi configurare il port forwarding sul router:

443 -> 443 server dove si trova Caddy
9000 -> 9000 server dove si trova Caddy

sul server dovrai avere:

Caddy in ascolto sulle porte 443 e 9000
Homeassistant sulla 8123

In questo modo, guardando il tuo configuration.yaml ed il Caddyfile che hai condiviso, HA ti può rispondere in HTTP solo in locale ed in HTTPS solo da remoto.

Se sei sicuro di aver fatto così, l'unica cosa che mi viene da pensare è che tu abbia sbagliato a fare il bind dei volumi da qualche parte.

**Reply by giacgiac** [reply]
tutto corretto come hai scritto tu, per bind dei volumi cosa intendi?

**Reply by snakuzzo** [reply]
Quando crei il container, farai il bind di directory e file locali per farli vedere appunto al container...quello intendo.

**Reply by giacgiac** [reply]
come da guida..

**Reply by snakuzzo** [reply]
Non so che dirti...sicuramente hai sbagliato da qualche parte. Ma senza metterci le mani diventa difficile.
Se vuoi una di queste sere, se mi contatti su Telegram, posso dare un'occhiata da remoto (con Teamviewer o Anydesk o altro)

**Reply by giacgiac** [reply]
sarebbe di grande aiuto..dimmi tu quando..

**Reply by giacgiac** [reply]
anche perché ho casa completamente bloccata, molte cose viaggiano solo con HA

**Reply by snakuzzo** [reply]
Non ne ho idea. Ripeto...contattami su Telegram e poi eventualmente ci si mette d'accordo lì

**Reply by giacgiac** [reply]
va benissimo come ti trovo su telegram?

**Reply by snakuzzo** [reply]
vieni sulla chat indomus e mi trovi (stesso nick)
